sdk/core/azure-core/src/main/java/com/azure/core/models/ResponseInnerError.java

```diff
@@ -59,19 +59,10 @@ public ResponseInnerError setInnerError(ResponseInnerError innerError) {
 
     @Override
     public JsonWriter toJson(JsonWriter jsonWriter) {
-        jsonWriter.writeStartObject();
-
-        jsonWriter.writeStringField("code", code)
-            .writeFieldName("innererror");
-
-        if (innerError != null) {
-            innerError.toJson(jsonWriter);
-        } else {
-            jsonWriter.writeNull();
-        }
-
-        // Always flush at the end of writing an object.
-        return jsonWriter.writeEndObject()
+        return jsonWriter.writeStartObject()
+            .writeStringField("code", code)
+            .writeJsonCapableField("innererror", innerError, false)
```

Comment:
the `writeJsonCapableField` sounds very odd (after `JsonCapable` - maybe this should just be called `Json`?).
in JSON we can have either primitive, object or array. Perhaps this should be called writeObjectField or writeNestedJson ? See how JDK names this https://docs.oracle.com/javaee/7/api/javax/json/JsonObject.html or gson for instance https://www.javadoc.io/doc/com.google.code.gson/gson/2.6.2/com/google/gson/JsonObject.html

Reply:
`JsonObject` has a better ring to it, and I know we talked about `JsonSerializable` in the past. I'm leaning more towards `JsonSerializable` as it leaves more room for interpretation if a class is able to serialize into multiple formats, whereas `JsonObject` sounds more definitive.

That said, I'm liking the idea to rename `writeJsonCapable*` to `writeJson*`.